ShellPkg/Application/Shell/Shell.c

```diff
@@ -1259,9 +1259,15 @@ LocateStartupScript (
     }
 
     InternalEfiShellSetEnv (L"homefilesystem", StartupScriptPath, TRUE);
+    // MU_CHANGE START: Ensure FileDevicePath is a FILEPATH_DEVICE_PATH prior to accessing PathName
+    if ((FileDevicePath->Type == MEDIA_VENDOR_DP) && (FileDevicePath->SubType == MEDIA_FILEPATH_DP)) {

```

Comment:
Just wanted to double check that you confirmed the type and subtype. I seem to recall type=`MEDIA_DEVICE_PATH` and subtype=`MEDIA_FILEPATH_DP` being used more often for file device paths.

Reply:
That is correct.  The code assumed the device path node as evidenced by the cast to FILEPATH_DEVICE_PATH.  It is a FILEPATH_DEVICE_PATH on "normal" systems that do not expose FV's as mapped file systems.  If you map FV's, and boot to the shell, the DEVIDE_PATH node is a FvGuid/FvName node, and casting that to a FILEPATH_DEVICE_PATH causes StrCat go run into the weeds.